### DISCUSSION TITLE
Unit testing?

### DISCUSSION THREAD

**Discussion by drmcclelland** [threenine/api-template#14] [General]
Sorry, working on my mobile device, so I didn’t yet get a chance to install the API template. However, I did not see any unit tests? Is that part of this template?

**Comment by garywoodfine** [comment]
Yes, the template comes with Unit testing project created for the API Solution template.  
It makes use of the xUnit, Moq, Shouldly and NBuilder.  In a few weeks, hopefully we will update the documentation on how we make use of it. However, if you're familiar with Unit testing concepts etc it should feel fairly intuitive.    The Cake.Build script is already configured to execute the Unit and Integration tests on build.
